internal/observability/logging/logging.go

```diff
@@ -93,45 +92,55 @@ func doInitLogging(ctx context.Context, level string) {
 		}))),
 	)
 
-	handleUSR1Signal(ctx, minLogLevel, &atomicLevel)
-}
-
-// setLevelForDuration sets the global log level to the given level for a given duration. Reverts to the original
-// level after the duration.
-func setLevelForDuration(level, originalLevel zapcore.Level, duration time.Duration, inProgress *atomic.Bool, atomicLevel *zap.AtomicLevel) {
-	log := zap.S().Named("logging")
-
-	log.Infof("Temporarily setting global log level to %s for %s", level, duration)
-	atomicLevel.SetLevel(level)
-
-	time.AfterFunc(duration, func() {
-		log.Infof("Reverting global log level to %s", originalLevel)
-		atomicLevel.SetLevel(originalLevel)
-		inProgress.Store(false)
-	})
+	if minLogLevel > zap.DebugLevel {
+		handleUSR1Signal(ctx, minLogLevel, &atomicLevel)
+	}
 }
 
-// handleUSR1Signal sets the log level to zapcore.DebugLevel for some duration in case syscall.SIGUSR1 received.
+// handleUSR1Signal temporarily sets the log level to debug when a SIGUSR1 signal is received.
 func handleUSR1Signal(ctx context.Context, originalLevel zapcore.Level, atomicLevel *zap.AtomicLevel) {
 	sigusr1 := make(chan os.Signal, 1)
 	signal.Notify(sigusr1, syscall.SIGUSR1)
 
 	go func() {
-		inProgress := atomic.Bool{}
+		inProgress := false
+		doneChan := make(chan struct{}, 1)
 		for {
 			select {
 			case <-ctx.Done():
 				return
 			case <-sigusr1:
-				if !inProgress.Load() {
-					inProgress.Store(true)
-					setLevelForDuration(zapcore.DebugLevel, originalLevel, tmpLogLevelDuration, &inProgress, atomicLevel)
+				if !inProgress {
+					inProgress = true
+					go setLogLevelForDuration(ctx, doneChan, originalLevel, atomicLevel)
```

Comment:
It might be quite nice if sending an additional USR1 while in progress would extend the time period rather than dropping the signal?